internal/api/impl/auth/auth.go

```diff
@@ -15,70 +15,55 @@ package auth
 
 import (
 	"errors"
+	"fmt"
 	"net/http"
 
 	"github.com/labstack/echo/v4"
 	"github.com/perses/perses/internal/api/interface/v1/user"
 	"github.com/perses/perses/internal/api/shared"
 	"github.com/perses/perses/internal/api/shared/crypto"
-	databaseModel "github.com/perses/perses/internal/api/shared/database/model"
 	"github.com/perses/perses/pkg/model/api"
+	"github.com/perses/perses/pkg/model/api/config"
 	"github.com/sirupsen/logrus"
 )
 
-type Endpoint struct {
-	dao          user.DAO
-	jwt          crypto.JWT
-	isAuthEnable bool
+type Endpoint interface {
```

Comment:
mmm a bit sad to redefine the interface just for that.

 I will propose something to make it simpler.

Reply:
I can't wait to see it merged thanks  :)